src/lib/phdi-building-blocks/phdi_building_blocks/standardize.py

```diff
@@ -59,18 +62,139 @@ def standardize_patient_name(
+        code_type (str): A string equal to 'alpha_2', 'alpha_3', or 'numeric' to specify
+                         which type of standard country identifier to generate.
+    Returns:
+        (generator): A generator to return standardized countries when it is iterated
+                     over.
+
+    We currently only handle extracting countries from patient resources, however as
+    need arises is the future funcionality can be expanded.
+    """
+    resource_type = resource.get("resource").get("resourceType")
+    if resource_type == "Patient":
+        for address in resource.get("resource").get("address"):
+            country = address.get("country")
+            yield standardize_country(country, code_type)
+
 
-    >>> standardize_phone("(555) 555-1212")
-    '5555551212'
+def standardize_phone(
+    raw: str, resource: dict = None, country_extractor: Callable = country_extractor
+) -> str:
+    """
+    Given a phone number and optionally an associated FHIR resource and country
+    extraction function if able to parse the phone number return it in the E.164
+    standardard international format. If the phone number is not parseable return none.
+
+    Phone number parsing process:
+    A maximum of three attemtps are made to parse any phone number.
+    First, we try to parse the phone number without any additional country information.
+    If this succeeds it must have provided in a standard inernational format which is
+    the ideal case. If the first attempt fails and a FHIR resource associated with the
+    phone number and a country extraction function are provided we make a second attempt
+    to parse the phone number using any countries extracted from the resource. Finally,
+    in the case where the second attempt fails for a resource and country extraction
+    function are not provided we make a final attempt to parse the number assuming it
+    is American.
+
+    Args:
+        raw (str): Raw phone number to be standardized.
+
+        resource (dict): FHIR resource that the raw phone originated from that possibly
+                         contains information indicating which country the phone number
+                         is from.
+
+        country_extractor (Callable): A function to parse a provided resource that
+                                      yields a generator to return the standard 2 letter
+                                      abreviation of each country found in the resource
+                                      when itterated over.
+    Returns:
+        standardized (str): The standardized phone number in E.164 format when the raw
+                            phone number was succesfully parsed and an emptry string
+                            otherwise.
     """
 
-    raw = [x for x in raw if x.isnumeric()]
-    raw = "".join(raw)
-    if len(raw) != 10:
-        raw = None
-    return raw
+    phone_number = None
+
+    # First Attempt - Assume complete international format provided.
```

Comment:
If you pass in the list of countries, you could then add a line like `countries = [None] + countries + ["US"]`, and then do the following:

```python
standardized = ""
for country in countries:
    try:
        phone_number = phonenumbers.parse(raw, country)
    except phonenumbers.phonenumberutil.NumberParseException:
        continue
if standardized != "":
    standardized = str(
        phoenumbers.format_number(
            phone_number, phonenumbers.PhoneNumberFormat.E164
        )
    )
return standardized
```

This would handle all three of your edge cases in a much more succinct manner.

Reply:
This is great. I'm going to slightly expand on this by setting the default value of countries to `[None, "US"]`. THis maintains the option for the user to not extract and provide a list of countries. In the case where the user does provide a list we'll do `countries = [None] + countries + ["US"]` as you suggest.

Reply:
Also, note that I forgot to include a `break` statement before the `except`. That'll be necessary since you want to dump out of the loop upon a successful parsing.